avro/src/main/java/com/fasterxml/jackson/dataformat/avro/schema/AvroSchemaHelper.java

```diff
@@ -196,6 +196,6 @@ protected static String getTypeId(JavaType type) {
         if (type.isPrimitive()) {
             return ClassUtil.wrapperType(type.getRawClass()).getName();
         }
-        return type.toCanonical();
+        return type.getRawClass().getName();
```

Comment:
Ah ok. It would be great to have a test for this, if you have seen such failure?